### PR TITLE
prov/efa: four commits

### PR DESCRIPTION
**Pull request by wzamazon**
This PR contain 4 commits that are needed for an upcoming change that use fi_read to copy data into GPU buffer.

### CODE REVIEW

**Comment by wzamazon** [comment]
bot:aws:retest

File: prov/efa/src/rxr/rxr_pkt_type_req.c [hunk 1]
```diff
@@ -827,18 +827,19 @@ ssize_t rxr_pkt_proc_matched_medium_rtm(struct rxr_ep *ep,
...
 {
-	struct rxr_pkt_entry *cur;
+	struct rxr_pkt_entry *cur, *nxt;
 	struct efa_mr *desc;
 	char *data;
 	size_t offset, hdr_size, data_size;
 
 	cur = pkt_entry;
 	while (cur) {
-		hdr_size = rxr_pkt_req_hdr_size(pkt_entry);
```
Comment (rwespetal) [inline prov/efa/src/rxr/rxr_pkt_type_req.c:837]: Is this a separate bug? Is it possible for the header sizes to be different across packets in a medium message?

File: prov/efa/src/rxr/rxr_ep.c [hunk 2]
```diff
@@ -1179,18 +1179,16 @@ int rxr_ep_init(struct rxr_ep *ep)
 		goto err_free_tx_pool;
```
Comment (rwespetal) [inline prov/efa/src/rxr/rxr_ep.c:1179]: Please fix this commit so it's authored by you.

File: prov/efa/src/rxr/rxr_pkt_entry.c [hunk 3]
```diff
@@ -201,16 +201,20 @@ void rxr_pkt_entry_copy(struct rxr_ep *ep,
...
+		new_entry_type);
 	assert(src->type == RXR_PKT_ENTRY_POSTED);
-	memcpy(dest, src, sizeof(struct rxr_pkt_entry));
-	memcpy(dest->pkt, src->pkt, ep->mtu_size);
 	dlist_init(&dest->entry);
 #if ENABLE_DEBUG
 	dlist_init(&dest->dbg_entry);
 #endif
-	dest->state = RXR_PKT_ENTRY_IN_USE;
+	dest->x_entry = src->x_entry;
```
Comment (rwespetal) [inline prov/efa/src/rxr/rxr_pkt_entry.c:209]: Can you add a comment that `dest->mr` is set in the `pkt_entry` alloc function?

File: prov/efa/src/rxr/rxr_pkt_type_req.c (hunk 1 above)
Comment (wzamazon) [reply prov/efa/src/rxr/rxr_pkt_type_req.c:837]: Yes, this is a bug. will split it out a bug fix commit.

File: prov/efa/src/rxr/rxr_pkt_entry.c (hunk 3 above)
Comment (wzamazon) [reply prov/efa/src/rxr/rxr_pkt_entry.c:209]: Sure.

File: prov/efa/src/rxr/rxr_ep.c (hunk 2 above)
Comment (wzamazon) [reply prov/efa/src/rxr/rxr_ep.c:1179]: Thanks. will fix

**Comment by wzamazon** [comment]
updated the pr with new revision.

**Pull request merged**